pkg/translate/translate.go

```diff
@@ -389,12 +392,48 @@ func (t *Translator) OverlayK8sSettings(yml string, icp *v1alpha2.IstioControlPl
 		if err != nil {
 			return "", fmt.Errorf("could not marshal to YAML in OverlayK8sSettings: %s", err)
 		}
-		mergedYAML, err := overlayK8s(baseYAML, overlayYAML, path[1:])
+
+		var dstNodeYaml, k8sSettingName string
+		if len(path) != 0 {
+			k8sSettingName = path[len(path)-1]
+		}
+
+		switch k8sSettingName {
+		case "env":
+			base := make(map[string]interface{})
+			if err := yaml.Unmarshal(baseYAML, &base); err != nil {
+				return "", fmt.Errorf("error: %s to unmarshal baseYAML:\n%s", err, baseYAML)
+			}
+
+			basePc, isBasePcFound, err := tpath.GetPathContext(base, path[1:])
+			if err != nil {
+				return "", err
+			}
+
+			if !isBasePcFound {
```

Comment:
I have a suggestion that may simplify the code and make it a bit more general. Instead of doing a special case for env, it should be possible to merge (rather than overwrite) each of the k8s destination subtrees inside overlayK8s.